common/persistence/persistenceTestBase.go

```diff
@@ -66,7 +68,8 @@ type (
 		MetadataManager     MetadataManager
 		VisibilityMgr       VisibilityManager
 		ShardInfo           *ShardInfo
-		ShardContext        *TestShardContext
+		TaskIDGenerator     TransferTaskIDGenerator
+		seqNum              int64
```

Comment:
Instead of seqNum, can we just have a default implementation which wraps seqNum logic behind the TransferTaskIDGenerator.  Then rest of the code does not need to check if TaskIDGenertor is set or not.